core/src/test/java/kafka/log/remote/RemoteLogManagerTest.java

```diff
@@ -2161,6 +2167,85 @@ RecordBatch findFirstBatch(RemoteLogInputStream remoteLogInputStream, long offse
         }
     }
 
+    @Test
+    public void testReadForFirstBatchInLogCompaction() throws RemoteStorageException, IOException {
+        FileInputStream fileInputStream = mock(FileInputStream.class);
+        RemoteLogInputStream remoteLogInputStream = mock(RemoteLogInputStream.class);
+        ClassLoaderAwareRemoteStorageManager rsmManager = mock(ClassLoaderAwareRemoteStorageManager.class);
+        RemoteLogSegmentMetadata segmentMetadata = mock(RemoteLogSegmentMetadata.class);
+        LeaderEpochFileCache cache = mock(LeaderEpochFileCache.class);
+        when(cache.epochForOffset(anyLong())).thenReturn(OptionalInt.of(1));
+
+        when(remoteStorageManager.fetchLogSegment(any(RemoteLogSegmentMetadata.class), anyInt()))
```

Comment:
There are two remote-storage-manager: `rsmManager` (local) and `remoteStorageManager` (global). Can we discard the latter one?

Reply:
done